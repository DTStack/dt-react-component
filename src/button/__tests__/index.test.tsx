import React from 'react';
import { UploadOutlined } from '@dtinsight/react-icons';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Button from '..';

describe('Button', () => {
    test('should support contentLayout success render', () => {
        const wrapper = render(<Button icon={<UploadOutlined />}>Primary</Button>);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders text correctly', () => {
        const { getByText } = render(<Button>Hello</Button>);
        expect(getByText('Hello')).toBeInTheDocument();
    });

    it('renders icon correctly', () => {
        const { container } = render(<Button icon={<UploadOutlined />} />);
        expect(container.querySelector('.dtc-button__icon')).toBeInTheDocument();
        expect(container.querySelector('.dtc-button__text')).not.toBeInTheDocument();
    });

    it('renders icon and text correctly', () => {
        const { getByText, container } = render(<Button icon={<UploadOutlined />}>Search</Button>);
        expect(getByText('Search')).toBeInTheDocument();
        expect(container.querySelector('.dtc-button__icon')).toBeInTheDocument();
    });

    it('applies custom className', () => {
        const { container } = render(<Button className="custom-class">Test</Button>);
        expect(container.firstChild).toHaveClass('custom-class');
    });

    it('passes other props to AntdButton', () => {
        const { getByText } = render(<Button type="primary">Primary</Button>);
        expect(getByText('Primary').parentNode).toHaveClass('ant-btn-primary');
    });

    it('applies size class to icon and text', () => {
        const { container } = render(
            <Button icon={<UploadOutlined />} size="small">
                Test
            </Button>
        );
        expect(container.querySelector('.dtc-button__icon--small')).toBeInTheDocument();
        expect(container.querySelector('.dtc-button__text--small')).toBeInTheDocument();
    });
});
