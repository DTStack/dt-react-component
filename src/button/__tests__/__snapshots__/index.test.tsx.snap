// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should support contentLayout success render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ant-btn ant-btn-default dtc-button"
        type="button"
      >
        <span
          class="dtc-button__icon dtc-button__icon--middle"
        >
          <span
            data-mock-icon="UploadOutlined"
          />
        </span>
        <span
          class="dtc-button__text dtc-button__text--middle"
        >
          Primary
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ant-btn ant-btn-default dtc-button"
      type="button"
    >
      <span
        class="dtc-button__icon dtc-button__icon--middle"
      >
        <span
          data-mock-icon="UploadOutlined"
        />
      </span>
      <span
        class="dtc-button__text dtc-button__text--middle"
      >
        Primary
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
